esquery.js

```diff
@@ -67,31 +67,32 @@
 
                 case 'matches':
                     for (i = 0, l = selector.selectors.length; i < l; ++i) {
-                        if (matches(node, selector.selectors[i], ancestry)) { return true; }
+                        if (matches(node, selector.selectors[i], ancestry, scope)) { return true; }
                     }
                     return false;
 
                 case 'compound':
                     for (i = 0, l = selector.selectors.length; i < l; ++i) {
-                        if (!matches(node, selector.selectors[i], ancestry)) { return false; }
+                        if (!matches(node, selector.selectors[i], ancestry, scope)) { return false; }
                     }
                     return true;
 
                 case 'not':
                     for (i = 0, l = selector.selectors.length; i < l; ++i) {
-                        if (matches(node, selector.selectors[i], ancestry)) { return false; }
+                        if (matches(node, selector.selectors[i], ancestry, scope)) { return false; }
                     }
                     return true;
 
                 case 'has':
-                    var a, collector = [];
+                    var a, collector = [], parent = ancestry[0];
```

Comment:
The naming here is a bit unclear as it's clobbered by the `parent` in the traverse method. Are they the same parent?